routes/index.js

```diff
@@ -19,4 +22,24 @@ router.get('/registration', function(req, res) {
   res.render('registration.njk');
 });
 
+router.post('/badgein',jsonParser,function(req,res){
+  if(!req.body)
+    { 
+     return res.sendStatus(400);
+    }
+  var BadgeNumber = req.body.badgeNumber;
+  if(typeof dbAPI.validateUser === undefined)
+   {
+     var url = '/registration' + req.body.badgeNumber;
+     res.redirect(url);
+   }
+  else
+  {
+   dbAPI.validateUser(undefined,BadgeNumber);
```

Comment:
You need to catch the return value of this function, if the value returned is `undefined` then you need to redirect to the registration page with the given badge ID. If the return value is a user object then you just need to say login was successful and just redirect back to the badgein page

Reply:
What U mean by returning user object ?, as the validateUser is returning true so the only value returned is true.

Reply:
I modified the code as u told me, please revise it and tell me ur opinion.

Reply:
Maybe the problem in the connection to the database anyway test it and tell me what is the problem.

Reply:
Actually I test the function, and I think everything works fine, can u check it ?